### ISSUE TITLE
[bug] Selecting a contact via widget, does not store parent_id

### ISSUE THREAD

**Issue by tobiasgraeber** [YetiForceCompany/YetiForceCRM#13722]
#13441 🐞 bug report

### ⁉️ Describe the bug

If you choose to select a contact via widget "external module" -> "contact" like here:

![grafik](https://user-images.githubusercontent.com/1668042/97873499-05e13d00-1d18-11eb-9fcc-a597a8eece0b.png)

Inside of the "Leads"-Module. The Widget does not get updated. The Widget "Updates" shows that a contact has been added to the leas - but the contact itself does not show up inside of the "contacts" Widget.

![grafik](https://user-images.githubusercontent.com/1668042/97873724-55276d80-1d18-11eb-8f31-ae564fc19b09.png)

If you add the relation between lead<->contact inside the contact-entry itself via the field "member of / parent_id" - it works.

![grafik](https://user-images.githubusercontent.com/1668042/97873929-915ace00-1d18-11eb-955d-4efcd7b26b7b.png)
![grafik](https://user-images.githubusercontent.com/1668042/97873886-843ddf00-1d18-11eb-9330-7e2e0494a21a.png)

### 🔥 How to trigger the error

1. Add a Widget for contacts inside of leads
2. Add a Contact to the contact list
3. Try to select that content inside of the lead via the contact widget
4. Selected contact should show up inside of the widget

### 👎 Actual Behavior

1. Add a Widget for contacts inside of leads
2. Add a Contact to the contact list
3. Try to select that content inside of the lead via the contact widget
4. Does not work / contact does not show up

### 👍 Expected Behavior

1. Add a Widget for contacts inside of leads
2. Add a Contact to the contact list
3. Try to select that content inside of the lead via the contact widget
4. Selected contact should show up inside of the widget

**Comment by tobiasgraeber** [comment]
Additional information:

I´ve clicked on the button "extended / erweitert" inside of the module contacts:

![grafik](https://user-images.githubusercontent.com/1668042/97877636-c1f13680-1d1d-11eb-93f0-d4d45ec11755.png)

Now, none of these two buttons are clickable anymore. Could that lead to the problem decribet above? Is there any way to just "reset" the contact modul?

I´ve also tried to create a own module, but sadly, for this, a "widget"-option doesn´t seem to exist by default?

....

**Comment by vovpff** [comment]
Look in community for Trebute.js bug. Maybe this is solution

**Comment by tobiasgraeber** [comment]
Thank you but... where to look and search exactly regarding this?

**Comment by vovpff** [comment]
#12732

**Comment by tobiasgraeber** [comment]
Thank you - but i think, thats another issue.
My Problem is not, that widgets are foldet. Thats working.

My Problem is, that, if i try to reference a contact via a widget, thats possible in the first step, but after assigning, the contact does NOT show up in the contact-widget.

Sadly i cannot reproduce that for examle in https://gitstable.yetiforce.com/index.php?module=Leads&view=Detail&record=268 because here in the test environment it seems as if you cannot "search" for a contact at the moment:

![grafik](https://user-images.githubusercontent.com/1668042/97895641-5bc3de00-1d34-11eb-94d1-5341affe8122.png)

Basically try to do that, select a contact, add it - it should show up in the "TEST"-Widgets list after that, but, on my side, it doesn´t at the moment. Note: "Add record" works, but i am talking about "Select record". :-)

**Comment by vovpff** [comment]
@tobiasgraeber i will try you issue tomorrow.

**Comment by vovpff** [comment]
@tobiasgraeber could you provide screenshots with english labels?

**Comment by tobiasgraeber** [comment]
Thanks - we can also have a teams meeting or some kind of videocall if you like. To me, the issue seems kind of easy but strange - i dont get any error but i also do not get any contact entry in the widget. Basically, as far as i can see, this ajax request is fired:

![grafik](https://user-images.githubusercontent.com/1668042/97900969-c7f61000-1d3b-11eb-9ad9-d92681bd2ee5.png)


Result ist:

![grafik](https://user-images.githubusercontent.com/1668042/97900985-cd535a80-1d3b-11eb-87d6-fba7315e3ca5.png)


But still the widget-contact-list does not show the entry:

![grafik](https://user-images.githubusercontent.com/1668042/97901033-df34fd80-1d3b-11eb-9e94-6ee5fdfbcf8b.png)


-----

Also i tried to debug \yetiforce\modules\Vtiger\actions\RelationAjax.php method "addRelation" but any output / debugging in here, does not have any effect - which file do i need to edit? Do i need to clear some cache or so? how?

..strange.

Thanks in advance!

**Comment by vovpff** [comment]
@tobiasgraeber i think you system works well. There is two general different type of relations availble (1:M and manytomany*).

For first one you need to input parent record name - Contact to Account, Contact to Lead and etc. The point of this type of relation is parent record has fixed relations with child records by field.

For manytomany you can add a lot of relations between different parent records and childs records. As example you can l relate same Contact for unlimeted value of Opportunities.

In your system you have active search button in widget what will be fixed (removed) for new release (5.4 or 6.0). This happens from time to time because system became improved with releases. Now you can diactivate this manually in database.

* I'm not sure what this is proper name of relation.

**Comment by tobiasgraeber** [comment]
@vovpff i understand. I also know i can disable the search. But thats not an improvement, its a huge step backwards then. For me it would be important to select a pre-existing contract via the search option and assign it, for example to a lead (from inside the lead details, via the widget itself! )

If that feature gets removed - what's the alternative???! 

If i need to go the other way around, so: creating the contact and assign it to a lead then from inside the contact - that would be much more inefficient

I understand the difference relation types - but nevertheless, there is no way for me to define the type that should be used, and even if, that does not help here :-) (At least i dont know where to do that) . 

So problem stays as it is - or, to ask in another way: what will be the "improved" solution / alternative for the search button then in the future release?

**Comment by tobiasgraeber** [comment]
Sorry, but i just dont get it. Let me ask one more question - for example on

 https://gitdeveloper.yetiforce.com/index.php?module=Leads&view=Detail&record=172&parent=47&mid=48

How can i add an EXISTING Contact to this lead?

**Comment by vovpff** [comment]
@tobiasgraeber https://gitdeveloper.yetiforce.com/index.php?module=Contacts&view=Detail&record=224&parent=47

**Comment by vovpff** [comment]
@tobiasgraeber now gitdev broken :(

**Comment by tobiasgraeber** [comment]
Ok, what to look for via this link :-) ? What do you mean?

**Comment by vovpff** [comment]
@tobiasgraeber this is answer for your question:
 if you want to relate existing Contact to this lead https://gitdeveloper.yetiforce.com/index.php?module=Leads&view=Detail&record=172&parent=47&mid=48
you need to fill Contact `Member of` field with name of lead like here https://gitdeveloper.yetiforce.com/index.php?module=Contacts&view=Detail&record=218&parent=47

**Comment by tobiasgraeber** [comment]
@vovpff Thanks again - yes thats one way. But i cannot understand, why the other way around gets removed? Thats so much more inefficient :-(

**Comment by tobiasgraeber** [comment]
@mariuszkrzaczkowski is this (removing of the search feature without alternative) really the case here?

**Comment by vovpff** [comment]
@tobiasgraeber I think what new way is better than old removed. This is difficult to describe

**Comment by tobiasgraeber** [comment]
Please try to describe or use sceenschots or a loom-video or so. I am really interested. 

- i understand your example above. But i cannot understand why the search option is getting removed.

I mean: If for example, somebody wants to assign multiple contacts to 1 lead, its way faster to use the search function via widget to find existing contacts and add them,insead of opening each contact one by one and assigning the lead inside of the details of a contact.

Cannot understand this decision. I mean, for products for example, its basically the same (search possible via widget - but here it works)

**Comment by vovpff** [comment]
@tobiasgraeber I think this is searching grows from old vtiger versions but now Ccontact should be hard related to Account/Lead. With this hard relation you can do more using workflow or handlers.

**Comment by rskrzypczak** [comment]
no activity, closing.
